docs/porting/libraries.md

```diff
@@ -1,5 +1,311 @@
+- Picking the correct project system in Visual Studio based on your needs
+- Porting your tests
+- Recommended approach for porting your code
+
+It's quite a lot of content, but porting is a task that will take some time, especially if you have a large codebase.  You should also be prepared to adapt the guidance here as needed to best fit your code.  Every codebase is different, so this article attempts to frame things in a flexible way, but you may find yourself needing to diverge from the prescribed guidance.
+
+## Prerequisites
+
+This article assumes you are using Visual Studio 2015 on Windows.
+
+This article also assumes that you have understood the [overview of the porting process](overview.md) and that you have resolved any issues with [3rd party dependencies](third-party-deps.md).  If you haven't done this already, it's highly recommended that you do that before moving forward.
+
+You may also wish to use [Reverse Package Search](https://packagesearch.azurewebsites.net), a service that will find packages where a given type is defined in.
+
+## Technologies Discontinued for .NET Core
+
+There are some technologies available for .NET Framework that you may use, but are outright unavailable for .NET Core.  These are technologies that we no longer promote for .NET Framework applications, and thus did not bring to .NET Core.
+
+### App Domains
+**Why was it discontinued?** AppDomains require runtime support and are generally quite expensive. While still implemented by CoreCLR, it’s not available in .NET Native and we don’t plan on adding this capability there.
+
+**What should you use instead?** AppDomains were used for different purposes. For code isolation, we recommend processes and/or containers. For dynamic loading of assemblies, we recommend the new [AssemblyLoadContext](http://dotnet.github.io/api/System.Runtime.Loader.AssemblyLoadContext.html) class.
+
+### Remoting
+**Why was it discontinued?** The idea of .NET remoting — which is transparent remote procedure calls — has been identified as a problematic architecture. Outside of that realm, it’s also used for cross AppDomain communication. On top of that, remoting requires runtime support and is quite heavyweight.
+
+**What should you use instead?** For communication across processes, inter-process communication (IPC) should be used, such as pipes or memory mapped files. Across machines, you should use a network based solution, preferably a low-overhead plain text protocol such as HTTP.
+
+### Binary serialization
+**Why was it discontinued?** After a decade of servicing, we’ve learned that serialization is incredibly complicated and a huge compatibility burden for the types supporting it. Thus, we made the decision that serialization should be a protocol implemented on top of the available public APIs. However, binary serialization requires intimate knowledge of the types because it allows to serialize object graphs, which includes private state.
```

Comment:
I don't understand why the last sentence begins with "however". It seems like what you're trying to say is that because binary serialization used private state, replacing it for types you don't own may be challenging?